client-java/src/main/java/uk/gov/gchq/palisade/client/download/DownloadManager.java

```diff
@@ -0,0 +1,225 @@
+
+    /**
+     * Create a new DownloadManager with the the provided
+     * {@code DownloadManagerSetup}
+     *
+     * @param downloadManagerSetup The setup
+     */
+    private DownloadManager(final DownloadManagerSetup downloadManagerSetup) {
+        this.setup = checkArgument(downloadManagerSetup);
+        this.buffer = new LinkedBlockingQueue<>();
+        this.executor = new ThreadPoolExecutor(
+            CORE_POOL_SIZE,
+            setup.getNumThreads(),
+            KEEP_ALIVE_SECONDS,
+            TimeUnit.SECONDS,
+            this.buffer);
+        executor.allowCoreThreadTimeOut(true);
+        LOGGER.debug("### Download manager created with thread pool size of {}", setup.getNumThreads());
+    }
+
+    /**
+     * Helper method to create a {@code DownloadManager} using a builder function
+     *
+     * @param func The builder function
+     * @return a newly created data request instance
+     */
+    @SuppressWarnings("java:S3242") // I REALLY want to use UnaryOperator here SonarQube!!!
+    public static DownloadManager createDownloadManager(final UnaryOperator<DownloadManagerSetup.Builder> func) {
+        return new DownloadManager(func.apply(DownloadManagerSetup.builder()).build());
+    }
+
+    /**
+     * Blocks until all tasks have completed execution after a shutdown request, or
+     * the timeout occurs, or the current thread is interrupted, whichever happens
+     * first.
+     *
+     * @param timeout the maximum time to wait
+     * @param unit    the time unit of the timeout argument
+     * @return true if this download manager terminated and false if the timeout
+     *         elapsed before termination
+     * @throws InterruptedException if interrupted while waiting
+     */
+    public boolean awaitTermination(final long timeout, final TimeUnit unit) throws InterruptedException {
+        LOGGER.debug("Awaiting for download manager to terminate for {} {} ", timeout, unit);
+        return executor.awaitTermination(timeout, unit);
+    }
+
+    @Override
+    public int getAvaliableSlots() {
+        return buffer.remainingCapacity();
+    }
+
+    /**
+     * Returns the download tracker for this download manager
+     *
+     * @return the download tracker for this download manager
+     */
+    public DownloadManagerStatus getDownloadTracker() {
+        return this;
+    }
+
+    @Override
+    public boolean hasAvailableSlots() {
+        return getAvaliableSlots() > 0;
+    }
+
+    /**
+     * Schedule the provided resource for download
+     *
+     * @param resource      The resource describing the data to be downloaded.
+     * @param eventbus      The event bus to be use when posting download events
+     * @param receiver      The receiver instance which will be passed the
+     *                      dowbnloaded input stream for processing
+     * @param configuration The job configuration
+     * @return the download id
+     */
+    @SuppressWarnings("java:S2221")
```

Comment:
remove suppress warnings

Reply:
I've removed this one as the `Downloader` called in the `try` catches `Exception` as well.